Chris

chochgesang4 opened the pull request:
Pull request of June 21st changes

chochgesang4 committed:
Merge pull request #5 from Crstt/main

update to current main

Crstt committed:
Merge pull request #8 from Crstt/main

Update matteo branch

chochgesang4 committed:
Merge pull request #10 from Crstt/main

catching up

christopherhochgesang committed:
-Fills out checkpoint manager
-Modifies 2 checkpoint types to include a name, removes some parameters
-includes a few tests for the checkpoint manager
-initialization of some checkpointers

christopherhochgesang committed:
actual commit from june 21
checkpoint manager controls checkpoints
adds gui based calls for town and river scenarios as well
adds terminal based calls for fallback/testing
small changes to constructors

includes a simple testing file

Crstt commented on checkpointManager.py:
```diff
@@ -1,9 +1,98 @@
+                    if(playerChoice==1):
+                        self.scenarioManager.callScenario(self.scenarioManager,"travel")
+                        loopTillValidInput=False
+                    elif(playerChoice==2):
```
Check supply is not necessary. Shows supply on screen

Crstt commented on checkpointManager.py:
```diff
@@ -1,9 +1,98 @@
+        self.scenarioManager = scenarioManager
     
+    #checkpoints are linear, so this will advance one checkpoint forward. if the checkpoint manager is already at the last checkpoint, call the gameover scenario
     def updateState(self):
```
- [x] Rename to nextCheckpoint
- [ ] Needs to update/return the distance to the next checkpoint in the player class
- [ ] Needs to return the updated money of the player (for ferry / food)
- [ ] Needs to return the amount of food bought

Crstt committed:
Update to scenario menager and scenarios

Crstt committed:
Call Scenario by name and modifiers

chochgesang4 committed:
Merge pull request #12 from Crstt/Matteo

Matteo

christopherhochgesang committed:
Fixes things from last commit
Adds a travel specialScenario and a buyFerry specialScenario.
both move the player 10 distance, buyferry checks for money first in checkpoint manager
